src/BlossomV/MinCostPerfectMatching.jl

```
@@ -0,0 +1,21 @@
+function min_cost_perfect_matching end
+# see https://github.com/mauro3/SimpleTraits.jl/issues/47#issuecomment-327880153 for syntax
+
+@traitfn function min_cost_perfect_matching(g::AG::(!IsDirected)) where {T<:Integer, AG <:AbstractGraph{T}}
+     nvg = nv(g)
+
+     if nvg%2 
```

```julia
     if nvg % 2 != 0 
         throw(ArgumentError("Odd number of vertices $nvg, no perfect matching"))
```

---

I wonder, if we really should throw an error in this case, or we could also calculate what in the literature is called (I think) a  near-perfect-matching.

---

The algorithm fails if a perfect matching doesn't exist, so it is important to throw an error at the start itself.